Supplementable name should not use ASCIILiteralPtrHash

rniwa opened the pull request:
#### 3612b1cc728638d08a16c14b515049747aeb8447
<pre>
Supplementable name should not use ASCIILiteralPtrHash
<a href="https://bugs.webkit.org/show_bug.cgi?id=272427">https://bugs.webkit.org/show_bug.cgi?id=272427</a>

Reviewed by Chris Dumez.

Use the default hash traits.

* Source/WebCore/platform/Supplementable.h:

Canonical link: <a href="https://commits.webkit.org/277286@main">https://commits.webkit.org/277286@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d08a55b0fc627dd9e145a2d0059fef7ab2f7bab2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/47206 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/26385 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/49853 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/49889 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/43255 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/49513 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/31729 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/23845 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/38454 "Failure limit exceed. At least found 1 new test failure: fast/css-custom-paint/out-of-memory-while-adding-worklet-module.html (failure)") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/47787 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/23881 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/40690 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/19767 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/21291 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/41855 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/5250 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/43588 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/42253 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/51765 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/22236 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/18606 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/45751 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/23508 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/44760 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/24295 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6642 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/23226 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/platform/Supplementable.h:
```diff
@@ -122,7 +122,7 @@ class Supplementable {
 
 private:
-    using SupplementMap = HashMap<ASCIILiteral, std::unique_ptr<Supplement<T>>, ASCIILiteralPtrHash>;
+    using SupplementMap = HashMap<ASCIILiteral, std::unique_ptr<Supplement<T>>>;
```
We may be able to get rid of the ASCIILiteralPtrHash implementation too.

rniwa replied:
Hm... it's still used in NetworkProcessSupplementMap, WebContextSupplementMap, and WebProcessSupplementMap. I guess we could move them off of this hash traits as well?

cdumez replied:
Yes, we should. They all are error prone in the same way.

rniwa replied:
Let me do that in a separate PR though.

rniwa replied:
https://github.com/WebKit/WebKit/pull/27059

rniwa committed:
Supplementable name should not use ASCIILiteralPtrHash
https://bugs.webkit.org/show_bug.cgi?id=272427

Reviewed by Chris Dumez.

Use the default hash traits.

* Source/WebCore/platform/Supplementable.h:

Canonical link: https://commits.webkit.org/277286@main
